src/binder/bind_expression/bind_null_operator_expression.cpp

```diff
@@ -12,7 +13,16 @@ std::shared_ptr<Expression> ExpressionBinder::bindNullOperatorExpression(
     const ParsedExpression& parsedExpression) {
     expression_vector children;
     for (auto i = 0u; i < parsedExpression.getNumChildren(); ++i) {
-        children.push_back(bindExpression(*parsedExpression.getChild(i)));
+        if (parsedExpression.getChild(i)->getExpressionType() == common::ExpressionType::LITERAL) {
```

Comment:
We probably can reuse existing code path to resolve ANY data type for NULL literals
```
auto boundChild = bindExpression(*parsedExpression.getChild(i));
// if boundChild.dataType == ANY
//     implicitCastIfNecessary
```

Reply:
I've applied the requested change in the new force push

Reply:
I end up making a minor change to refactor the code.